### ISSUE TITLE
Support switching RemoteModule between train/eval

### ISSUE THREAD

**Issue by SciPioneer** [pytorch/pytorch#51480]
## 🚀 Feature
Currently [RemoteModule](https://github.com/pytorch/pytorch/blob/master/torch/distributed/nn/api/remote_module.py#L68) does not support [train](https://github.com/pytorch/pytorch/blob/master/torch/distributed/nn/api/remote_module.py#L341) or [eval](https://github.com/pytorch/pytorch/blob/master/torch/distributed/nn/api/remote_module.py#L344). However, some behaviors like `Dropout` and `BatchNorm` in these two modes should work differently. Recently, `SyncBatchNorm` also behaves differently (#56982) between train/eval mode.

These methods should call the underlying train/eval methods on the actual nn.Module: https://github.com/pytorch/pytorch/blob/master/torch/nn/modules/module.py#L1622

cc @pietern @mrshenli @pritamdamania87 @zhaojuanmao @satgera @gqchen @aazzolini @rohan-varma @jjlilley @osalpekar @jiayisuse @agolynski @SciPioneer @H-Huang @mrzzd @cbalioglu
